### PR TITLE
Make IContainer et all typed

### PR DESCRIPTION
**Pull request by rullzer**
@MorrisJobke found this branch of mine lying around.

Lets see what CI thinks

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
Code looks good

**Review by MorrisJobke — APPROVED** [review]

File: lib/private/AppFramework/Utility/SimpleContainer.php [hunk 1]
```diff
@@ -90,7 +91,7 @@ private function buildClass(ReflectionClass $class) {
 	 * @return \stdClass
 	 * @throws QueryException if the class could not be found or instantiated
 	 */
-	public function resolve($name) {
+	public function resolve(string $name): \stdClass {
```
Comment (MorrisJobke) [inline lib/private/AppFramework/Utility/SimpleContainer.php]: 💥 💥 💥 
TypeError: Return value of OC\AppFramework\Utility\SimpleContainer::resolve() must be an instance of stdClass, instance of OC\Support\CrashReport\Registry returned in

**Comment by ChristophWurst** [comment]
```
1) Test\AppFramework\DependencyInjection\DIIntergrationTests::testInjectFromServer
TypeError: Return value of OC\AppFramework\DependencyInjection\DIContainer::getServer() must implement interface OCP\IServerContainer, instance of OC\ServerContainer returned
```
I love this class hierarchy.

`\OC\ServerContanier` extends `\OC\SimpleContainer`
`\OC\SimpleContanier` implements `\OCP\IContainer`
`\OCP\IServerContainer` extends `\OCP\IContainer`
`\OC\Server` extends `OC\ServerContainer` and implments `\IServerContainer`

so `DIContainer::getServer` should either just return `IContainer` or we have to change the class/interface hierarchy.

**Comment by rullzer** [comment]
Yeah it is messy... :speak_no_evil: 

I think this test is failing because we do sneaky things in the test. I'll check it out

**Comment by MorrisJobke** [comment]
@rullzer I guess nothing for 15, right?

**Comment by rullzer** [comment]
Correct

**Pull request closed without merging**